api/types/types_errors.go

```diff
@@ -51,3 +53,34 @@ type ErrBatchProcess struct{ goof.Goof }
 // ErrBadFilter occurs when a bad filter is supplied via the filter query
 // string.
 type ErrBadFilter struct{ goof.Goof }
+
+// ErrMissingStorageService occurs when the storage service is expected in
+// the provided context but is not there.
+var ErrMissingStorageService = errors.New("missing storage service")
```

Comment:
Why the use of `errors` instead of `goof` here? This one is defined differently than `ErroTimedOut` and `ErrNotImplemented`